flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/NOTICE

```diff
@@ -0,0 +1,45 @@
+flink-sql-connector-elasticsearch7
```

Comment:
why does this have a NOTICE file while the ES 6 connector doesn't?

Reply:
Hi @aljoscha This NOTICE file referenced `flink-sql-connector-elasticsearch6`. The `flink-connector-elasticsearch7` referenced `flink-connector-elasticsrearch6`, so they both have no NOTICE file.

Reply:
Did you check whether the contents of the new `NOTICE` file is correct? A quick `mvn dependency:tree` on the new sql connector shows some dependencies that are not in the `NOTICE` file. I think these are new compared to ES6, which didn't have them. You can compare the output of `mvn depdendency:tree` on the ES6 sql jar and on the new ES7 sql jar and also compare to the `NOTICE` files.

Reply:
Sorry, I forgot to check the dependency list. I will do the comparation soon.

Reply:
@aljoscha I have run `mvn depdendency:tree` command and checked the license of every dependency through maven repository. Can you review again?

Reply:
Quite some of the dependencies are test dependencies or transitive dependencies of test dependencies, I don't think they should be added here.

For some other dependencies, I only noticed this now but the maven shade configuration excludes some dependencies, so they also should not be added here. I'm also wondering which of the newly added dependencies to exclude here. Maybe @twalthr, who created the first versions of the ES6 table connector can shed some light on this.

Reply:
> Quite some of the dependencies are test dependencies or transitive dependencies of test dependencies, I don't think they should be added here.

Sorry, I do not know we should exclude test dependencies before adding missing dependencied into the notice file. Will remove them when we can decide excluded dependencied list.

Besides the license issue, I want to know whether there is any other issue need to be processed? If there is, you can point out so that I can fixed it and at the same time we can wait reply from @twalthr .

Reply:
I don't think there is another issue! 👌

Reply:
All dependencies of `flink-sql-connector-elasticsearch7_2.11` listed below:

```
[INFO] org.apache.flink:flink-sql-connector-elasticsearch7_2.11:jar:1.10-SNAPSHOT
[INFO] +- org.apache.flink:flink-connector-elasticsearch-base_2.11:jar:1.10-SNAPSHOT:compile
[INFO] +- org.apache.flink:flink-connector-elasticsearch7_2.11:jar:1.10-SNAPSHOT:compile
[INFO] |  \- org.elasticsearch.client:elasticsearch-rest-high-level-client:jar:7.3.2:compile
[INFO] |     +- org.elasticsearch:elasticsearch:jar:7.3.2:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-core:jar:7.3.2:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-secure-sm:jar:7.3.2:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-x-content:jar:7.3.2:compile
[INFO] |     |  |  +- org.yaml:snakeyaml:jar:1.17:compile
[INFO] |     |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.8.11:compile
[INFO] |     |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-smile:jar:2.8.11:compile
[INFO] |     |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.8.11:compile
[INFO] |     |  |  \- com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:jar:2.8.11:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-geo:jar:7.3.2:compile
[INFO] |     |  +- org.apache.lucene:lucene-core:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-analyzers-common:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-backward-codecs:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-grouping:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-highlighter:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-join:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-memory:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-misc:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-queries:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-queryparser:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-sandbox:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-spatial:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-spatial-extras:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-spatial3d:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-suggest:jar:8.1.0:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-cli:jar:7.3.2:compile
[INFO] |     |  |  \- net.sf.jopt-simple:jopt-simple:jar:5.0.2:compile
[INFO] |     |  +- com.carrotsearch:hppc:jar:0.8.1:compile
[INFO] |     |  +- joda-time:joda-time:jar:2.5:compile
[INFO] |     |  +- com.tdunning:t-digest:jar:3.2:compile
[INFO] |     |  +- org.hdrhistogram:HdrHistogram:jar:2.1.9:compile
[INFO] |     |  +- org.apache.logging.log4j:log4j-api:jar:2.11.1:compile
[INFO] |     |  \- org.elasticsearch:jna:jar:4.5.1:compile
[INFO] |     +- org.elasticsearch.client:elasticsearch-rest-client:jar:7.3.2:compile
[INFO] |     |  +- org.apache.httpcomponents:httpclient:jar:4.5.3:compile
[INFO] |     |  +- org.apache.httpcomponents:httpcore:jar:4.4.6:compile
[INFO] |     |  +- org.apache.httpcomponents:httpasyncclient:jar:4.1.4:compile
[INFO] |     |  +- org.apache.httpcomponents:httpcore-nio:jar:4.4.11:compile
[INFO] |     |  +- commons-codec:commons-codec:jar:1.10:compile
[INFO] |     |  \- commons-logging:commons-logging:jar:1.1.3:compile
[INFO] |     +- org.elasticsearch.plugin:parent-join-client:jar:7.3.2:compile
[INFO] |     +- org.elasticsearch.plugin:aggs-matrix-stats-client:jar:7.3.2:compile
[INFO] |     +- org.elasticsearch.plugin:rank-eval-client:jar:7.3.2:compile
[INFO] |     \- org.elasticsearch.plugin:lang-mustache-client:jar:7.3.2:compile
[INFO] |        \- com.github.spullara.mustache.java:compiler:jar:0.9.3:compile
[INFO] +- org.apache.flink:flink-table-api-java-bridge_2.11:jar:1.10-SNAPSHOT:provided (optional) 
[INFO] |  +- org.apache.flink:flink-table-api-java:jar:1.10-SNAPSHOT:provided
[INFO] |  +- org.apache.flink:flink-java:jar:1.10-SNAPSHOT:provided
[INFO] |  |  +- org.apache.flink:flink-core:jar:1.10-SNAPSHOT:provided
[INFO] |  |  |  +- org.apache.flink:flink-annotations:jar:1.10-SNAPSHOT:provided
[INFO] |  |  |  +- org.apache.flink:flink-metrics-core:jar:1.10-SNAPSHOT:provided
[INFO] |  |  |  +- com.esotericsoftware.kryo:kryo:jar:2.24.0:provided
[INFO] |  |  |  |  \- com.esotericsoftware.minlog:minlog:jar:1.2:provided
[INFO] |  |  |  +- commons-collections:commons-collections:jar:3.2.2:provided
[INFO] |  |  |  \- org.apache.commons:commons-compress:jar:1.18:provided
[INFO] |  |  +- org.apache.flink:flink-shaded-asm-7:jar:7.1-8.0:provided
[INFO] |  |  +- org.apache.commons:commons-lang3:jar:3.3.2:provided
[INFO] |  |  \- org.apache.commons:commons-math3:jar:3.5:provided
[INFO] |  \- org.apache.flink:flink-streaming-java_2.11:jar:1.10-SNAPSHOT:provided
[INFO] |     +- org.apache.flink:flink-runtime_2.11:jar:1.10-SNAPSHOT:provided
[INFO] |     |  +- org.apache.flink:flink-queryable-state-client-java:jar:1.10-SNAPSHOT:provided
[INFO] |     |  +- org.apache.flink:flink-hadoop-fs:jar:1.10-SNAPSHOT:provided
[INFO] |     |  +- commons-io:commons-io:jar:2.4:provided
[INFO] |     |  +- org.apache.flink:flink-shaded-netty:jar:4.1.39.Final-8.0:provided
[INFO] |     |  +- org.apache.flink:flink-shaded-jackson:jar:2.9.8-8.0:provided
[INFO] |     |  +- commons-cli:commons-cli:jar:1.3.1:provided
[INFO] |     |  +- org.javassist:javassist:jar:3.24.0-GA:provided
[INFO] |     |  +- org.scala-lang:scala-library:jar:2.11.12:provided
[INFO] |     |  +- com.typesafe.akka:akka-actor_2.11:jar:2.5.21:provided
[INFO] |     |  |  +- com.typesafe:config:jar:1.3.0:provided
[INFO] |     |  |  \- org.scala-lang.modules:scala-java8-compat_2.11:jar:0.7.0:provided
[INFO] |     |  +- com.typesafe.akka:akka-stream_2.11:jar:2.5.21:provided
[INFO] |     |  |  +- org.reactivestreams:reactive-streams:jar:1.0.2:provided
[INFO] |     |  |  \- com.typesafe:ssl-config-core_2.11:jar:0.3.7:provided
[INFO] |     |  +- com.typesafe.akka:akka-protobuf_2.11:jar:2.5.21:provided
[INFO] |     |  +- com.typesafe.akka:akka-slf4j_2.11:jar:2.5.21:provided
[INFO] |     |  +- org.clapper:grizzled-slf4j_2.11:jar:1.3.2:provided
[INFO] |     |  +- com.github.scopt:scopt_2.11:jar:3.5.0:provided
[INFO] |     |  +- org.xerial.snappy:snappy-java:jar:1.1.4:provided
[INFO] |     |  \- com.twitter:chill_2.11:jar:0.7.6:provided
[INFO] |     |     \- com.twitter:chill-java:jar:0.7.6:provided
[INFO] |     +- org.apache.flink:flink-clients_2.11:jar:1.10-SNAPSHOT:provided
[INFO] |     |  \- org.apache.flink:flink-optimizer_2.11:jar:1.10-SNAPSHOT:provided
[INFO] |     \- org.apache.flink:flink-shaded-guava:jar:18.0-8.0:provided
[INFO] +- org.apache.flink:flink-connector-elasticsearch-base_2.11:test-jar:tests:1.10-SNAPSHOT:test
[INFO] +- org.apache.flink:flink-json:jar:1.10-SNAPSHOT:test
[INFO] +- org.apache.flink:flink-table-planner_2.11:test-jar:tests:1.10-SNAPSHOT:test
[INFO] |  +- org.apache.flink:flink-table-common:jar:1.10-SNAPSHOT:provided
[INFO] |  \- org.apache.flink:flink-table-api-scala-bridge_2.11:jar:1.10-SNAPSHOT:test
[INFO] |     +- org.apache.flink:flink-table-api-scala_2.11:jar:1.10-SNAPSHOT:test
[INFO] |     |  +- org.scala-lang:scala-reflect:jar:2.11.12:test
[INFO] |     |  \- org.scala-lang:scala-compiler:jar:2.11.12:test
[INFO] |     |     +- org.scala-lang.modules:scala-xml_2.11:jar:1.0.5:test
[INFO] |     |     \- org.scala-lang.modules:scala-parser-combinators_2.11:jar:1.1.1:provided
[INFO] |     +- org.apache.flink:flink-scala_2.11:jar:1.10-SNAPSHOT:test
[INFO] |     \- org.apache.flink:flink-streaming-scala_2.11:jar:1.10-SNAPSHOT:test
[INFO] +- org.slf4j:slf4j-api:jar:1.7.15:provided
[INFO] +- com.google.code.findbugs:jsr305:jar:1.3.9:provided
[INFO] +- org.apache.flink:flink-test-utils-junit:jar:1.10-SNAPSHOT:test
[INFO] +- org.apache.flink:force-shading:jar:1.10-SNAPSHOT:compile
[INFO] +- junit:junit:jar:4.12:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] +- org.mockito:mockito-core:jar:2.21.0:test
[INFO] |  +- net.bytebuddy:byte-buddy:jar:1.8.15:test
[INFO] |  +- net.bytebuddy:byte-buddy-agent:jar:1.8.15:test
[INFO] |  \- org.objenesis:objenesis:jar:2.1:provided
[INFO] +- org.powermock:powermock-module-junit4:jar:2.0.2:test
[INFO] |  \- org.powermock:powermock-module-junit4-common:jar:2.0.2:test
[INFO] |     +- org.powermock:powermock-reflect:jar:2.0.2:test
[INFO] |     \- org.powermock:powermock-core:jar:2.0.2:test
[INFO] +- org.powermock:powermock-api-mockito2:jar:2.0.2:test
[INFO] |  \- org.powermock:powermock-api-support:jar:2.0.2:test
[INFO] +- org.hamcrest:hamcrest-all:jar:1.3:test
[INFO] +- org.slf4j:slf4j-log4j12:jar:1.7.15:test
[INFO] \- log4j:log4j:jar:1.2.17:test
```

Removed test dependencies and transitive dependencies of test dependencies, remain dependencies:

```
[INFO] org.apache.flink:flink-sql-connector-elasticsearch7_2.11:jar:1.10-SNAPSHOT
[INFO] +- org.apache.flink:flink-connector-elasticsearch-base_2.11:jar:1.10-SNAPSHOT:compile
[INFO] +- org.apache.flink:flink-connector-elasticsearch7_2.11:jar:1.10-SNAPSHOT:compile
[INFO] |  \- org.elasticsearch.client:elasticsearch-rest-high-level-client:jar:7.3.2:compile
[INFO] |     +- org.elasticsearch:elasticsearch:jar:7.3.2:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-core:jar:7.3.2:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-secure-sm:jar:7.3.2:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-x-content:jar:7.3.2:compile
[INFO] |     |  |  +- org.yaml:snakeyaml:jar:1.17:compile
[INFO] |     |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.8.11:compile
[INFO] |     |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-smile:jar:2.8.11:compile
[INFO] |     |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.8.11:compile
[INFO] |     |  |  \- com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:jar:2.8.11:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-geo:jar:7.3.2:compile
[INFO] |     |  +- org.apache.lucene:lucene-core:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-analyzers-common:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-backward-codecs:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-grouping:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-highlighter:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-join:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-memory:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-misc:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-queries:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-queryparser:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-sandbox:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-spatial:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-spatial-extras:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-spatial3d:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-suggest:jar:8.1.0:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-cli:jar:7.3.2:compile
[INFO] |     |  |  \- net.sf.jopt-simple:jopt-simple:jar:5.0.2:compile
[INFO] |     |  +- com.carrotsearch:hppc:jar:0.8.1:compile
[INFO] |     |  +- joda-time:joda-time:jar:2.5:compile
[INFO] |     |  +- com.tdunning:t-digest:jar:3.2:compile
[INFO] |     |  +- org.hdrhistogram:HdrHistogram:jar:2.1.9:compile
[INFO] |     |  +- org.apache.logging.log4j:log4j-api:jar:2.11.1:compile
[INFO] |     |  \- org.elasticsearch:jna:jar:4.5.1:compile
[INFO] |     +- org.elasticsearch.client:elasticsearch-rest-client:jar:7.3.2:compile
[INFO] |     |  +- org.apache.httpcomponents:httpclient:jar:4.5.3:compile
[INFO] |     |  +- org.apache.httpcomponents:httpcore:jar:4.4.6:compile
[INFO] |     |  +- org.apache.httpcomponents:httpasyncclient:jar:4.1.4:compile
[INFO] |     |  +- org.apache.httpcomponents:httpcore-nio:jar:4.4.11:compile
[INFO] |     |  +- commons-codec:commons-codec:jar:1.10:compile
[INFO] |     |  \- commons-logging:commons-logging:jar:1.1.3:compile
[INFO] |     +- org.elasticsearch.plugin:parent-join-client:jar:7.3.2:compile
[INFO] |     +- org.elasticsearch.plugin:aggs-matrix-stats-client:jar:7.3.2:compile
[INFO] |     +- org.elasticsearch.plugin:rank-eval-client:jar:7.3.2:compile
[INFO] |     \- org.elasticsearch.plugin:lang-mustache-client:jar:7.3.2:compile
[INFO] |        \- com.github.spullara.mustache.java:compiler:jar:0.9.3:compile
[INFO] +- org.apache.flink:flink-table-api-java-bridge_2.11:jar:1.10-SNAPSHOT:provided (optional) 
[INFO] |  +- org.apache.flink:flink-table-api-java:jar:1.10-SNAPSHOT:provided
[INFO] |  +- org.apache.flink:flink-java:jar:1.10-SNAPSHOT:provided
[INFO] |  |  +- org.apache.flink:flink-core:jar:1.10-SNAPSHOT:provided
[INFO] |  |  |  +- org.apache.flink:flink-annotations:jar:1.10-SNAPSHOT:provided
[INFO] |  |  |  +- org.apache.flink:flink-metrics-core:jar:1.10-SNAPSHOT:provided
[INFO] |  |  |  +- com.esotericsoftware.kryo:kryo:jar:2.24.0:provided
[INFO] |  |  |  |  \- com.esotericsoftware.minlog:minlog:jar:1.2:provided
[INFO] |  |  |  +- commons-collections:commons-collections:jar:3.2.2:provided
[INFO] |  |  |  \- org.apache.commons:commons-compress:jar:1.18:provided
[INFO] |  |  +- org.apache.flink:flink-shaded-asm-7:jar:7.1-8.0:provided
[INFO] |  |  +- org.apache.commons:commons-lang3:jar:3.3.2:provided
[INFO] |  |  \- org.apache.commons:commons-math3:jar:3.5:provided
[INFO] |  \- org.apache.flink:flink-streaming-java_2.11:jar:1.10-SNAPSHOT:provided
[INFO] |     +- org.apache.flink:flink-runtime_2.11:jar:1.10-SNAPSHOT:provided
[INFO] |     |  +- org.apache.flink:flink-queryable-state-client-java:jar:1.10-SNAPSHOT:provided
[INFO] |     |  +- org.apache.flink:flink-hadoop-fs:jar:1.10-SNAPSHOT:provided
[INFO] |     |  +- commons-io:commons-io:jar:2.4:provided
[INFO] |     |  +- org.apache.flink:flink-shaded-netty:jar:4.1.39.Final-8.0:provided
[INFO] |     |  +- org.apache.flink:flink-shaded-jackson:jar:2.9.8-8.0:provided
[INFO] |     |  +- commons-cli:commons-cli:jar:1.3.1:provided
[INFO] |     |  +- org.javassist:javassist:jar:3.24.0-GA:provided
[INFO] |     |  +- org.scala-lang:scala-library:jar:2.11.12:provided
[INFO] |     |  +- com.typesafe.akka:akka-actor_2.11:jar:2.5.21:provided
[INFO] |     |  |  +- com.typesafe:config:jar:1.3.0:provided
[INFO] |     |  |  \- org.scala-lang.modules:scala-java8-compat_2.11:jar:0.7.0:provided
[INFO] |     |  +- com.typesafe.akka:akka-stream_2.11:jar:2.5.21:provided
[INFO] |     |  |  +- org.reactivestreams:reactive-streams:jar:1.0.2:provided
[INFO] |     |  |  \- com.typesafe:ssl-config-core_2.11:jar:0.3.7:provided
[INFO] |     |  +- com.typesafe.akka:akka-protobuf_2.11:jar:2.5.21:provided
[INFO] |     |  +- com.typesafe.akka:akka-slf4j_2.11:jar:2.5.21:provided
[INFO] |     |  +- org.clapper:grizzled-slf4j_2.11:jar:1.3.2:provided
[INFO] |     |  +- com.github.scopt:scopt_2.11:jar:3.5.0:provided
[INFO] |     |  +- org.xerial.snappy:snappy-java:jar:1.1.4:provided
[INFO] |     |  \- com.twitter:chill_2.11:jar:0.7.6:provided
[INFO] |     |     \- com.twitter:chill-java:jar:0.7.6:provided
[INFO] |     +- org.apache.flink:flink-clients_2.11:jar:1.10-SNAPSHOT:provided
[INFO] |     |  \- org.apache.flink:flink-optimizer_2.11:jar:1.10-SNAPSHOT:provided
[INFO] |     \- org.apache.flink:flink-shaded-guava:jar:18.0-8.0:provided
[INFO] +- org.slf4j:slf4j-api:jar:1.7.15:provided
[INFO] +- com.google.code.findbugs:jsr305:jar:1.3.9:provided
[INFO] +- org.apache.flink:force-shading:jar:1.10-SNAPSHOT:compile
```

Here is [flink shaded dependencies list](https://mvnrepository.com/search?q=flink-shaded) can be searched from maven repository. I do not know if it is complete. If there missed anything please let me know.

So removed shaded dependencies and dependencies with`org.apache.flink` prefix, remain dependencies list here:

```

[INFO] |  \- org.elasticsearch.client:elasticsearch-rest-high-level-client:jar:7.3.2:compile
[INFO] |     +- org.elasticsearch:elasticsearch:jar:7.3.2:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-core:jar:7.3.2:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-secure-sm:jar:7.3.2:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-x-content:jar:7.3.2:compile
[INFO] |     |  |  +- org.yaml:snakeyaml:jar:1.17:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-geo:jar:7.3.2:compile
[INFO] |     |  +- org.apache.lucene:lucene-core:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-analyzers-common:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-backward-codecs:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-grouping:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-highlighter:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-join:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-memory:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-misc:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-queries:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-queryparser:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-sandbox:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-spatial:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-spatial-extras:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-spatial3d:jar:8.1.0:compile
[INFO] |     |  +- org.apache.lucene:lucene-suggest:jar:8.1.0:compile
[INFO] |     |  +- org.elasticsearch:elasticsearch-cli:jar:7.3.2:compile
[INFO] |     |  |  \- net.sf.jopt-simple:jopt-simple:jar:5.0.2:compile
[INFO] |     |  +- com.carrotsearch:hppc:jar:0.8.1:compile
[INFO] |     |  +- joda-time:joda-time:jar:2.5:compile
[INFO] |     |  +- com.tdunning:t-digest:jar:3.2:compile
[INFO] |     |  +- org.hdrhistogram:HdrHistogram:jar:2.1.9:compile
[INFO] |     |  +- org.apache.logging.log4j:log4j-api:jar:2.11.1:compile
[INFO] |     |  \- org.elasticsearch:jna:jar:4.5.1:compile
[INFO] |     +- org.elasticsearch.client:elasticsearch-rest-client:jar:7.3.2:compile
[INFO] |     |  +- org.apache.httpcomponents:httpclient:jar:4.5.3:compile
[INFO] |     |  +- org.apache.httpcomponents:httpcore:jar:4.4.6:compile
[INFO] |     |  +- org.apache.httpcomponents:httpasyncclient:jar:4.1.4:compile
[INFO] |     |  +- org.apache.httpcomponents:httpcore-nio:jar:4.4.11:compile
[INFO] |     |  +- commons-codec:commons-codec:jar:1.10:compile
[INFO] |     |  \- commons-logging:commons-logging:jar:1.1.3:compile
[INFO] |     +- org.elasticsearch.plugin:parent-join-client:jar:7.3.2:compile
[INFO] |     +- org.elasticsearch.plugin:aggs-matrix-stats-client:jar:7.3.2:compile
[INFO] |     +- org.elasticsearch.plugin:rank-eval-client:jar:7.3.2:compile
[INFO] |     \- org.elasticsearch.plugin:lang-mustache-client:jar:7.3.2:compile
[INFO] |        \- com.github.spullara.mustache.java:compiler:jar:0.9.3:compile
[INFO] |  |  |  +- com.esotericsoftware.kryo:kryo:jar:2.24.0:provided
[INFO] |  |  |  |  \- com.esotericsoftware.minlog:minlog:jar:1.2:provided
[INFO] |  |  |  +- commons-collections:commons-collections:jar:3.2.2:provided
[INFO] |  |  |  \- org.apache.commons:commons-compress:jar:1.18:provided
[INFO] |  |  +- org.apache.commons:commons-lang3:jar:3.3.2:provided
[INFO] |  |  \- org.apache.commons:commons-math3:jar:3.5:provided
[INFO] |     |  +- commons-io:commons-io:jar:2.4:provided
[INFO] |     |  +- commons-cli:commons-cli:jar:1.3.1:provided
[INFO] |     |  +- org.javassist:javassist:jar:3.24.0-GA:provided
[INFO] |     |  +- org.scala-lang:scala-library:jar:2.11.12:provided
[INFO] |     |  +- com.typesafe.akka:akka-actor_2.11:jar:2.5.21:provided
[INFO] |     |  |  +- com.typesafe:config:jar:1.3.0:provided
[INFO] |     |  |  \- org.scala-lang.modules:scala-java8-compat_2.11:jar:0.7.0:provided
[INFO] |     |  +- com.typesafe.akka:akka-stream_2.11:jar:2.5.21:provided
[INFO] |     |  |  +- org.reactivestreams:reactive-streams:jar:1.0.2:provided
[INFO] |     |  |  \- com.typesafe:ssl-config-core_2.11:jar:0.3.7:provided
[INFO] |     |  +- com.typesafe.akka:akka-protobuf_2.11:jar:2.5.21:provided
[INFO] |     |  +- com.typesafe.akka:akka-slf4j_2.11:jar:2.5.21:provided
[INFO] |     |  +- org.clapper:grizzled-slf4j_2.11:jar:1.3.2:provided
[INFO] |     |  +- com.github.scopt:scopt_2.11:jar:3.5.0:provided
[INFO] |     |  +- org.xerial.snappy:snappy-java:jar:1.1.4:provided
[INFO] |     |  \- com.twitter:chill_2.11:jar:0.7.6:provided
[INFO] |     |     \- com.twitter:chill-java:jar:0.7.6:provided
[INFO] +- org.slf4j:slf4j-api:jar:1.7.15:provided
[INFO] +- com.google.code.findbugs:jsr305:jar:1.3.9:provided
```

So, do you think the above dependencies could be considered as the correct comparison set(to see whether their licences are under `ASL 2.0 ` or not?).

Reply:
You also have to remove those that are excluded in the `maven-shade-plugin` configuration, for example joda-time.

Reply:
@aljoscha I have addressed your comments and updated the PR. Please see the latest commit, if there is anything wrong, please let me know. Thanks.

Reply:
@twalthr Any other additional remarks?

Reply:
Hi @aljoscha It seems @twalthr is busy recently? Can you talk with him offline to see if there is anything need to be considered?

Reply:
Hi, this is not correct, the list of `flink-shaded` dependencies has nothing to do with the local shading in this connector. What I did now is to look at the output of `mvn clean install` in the ES7 connector directory. It has a section about what is included and what is not included. Based on this information I added some additional exclusions and I updated the NOTICE file: https://github.com/aljoscha/flink/commit/b80c8b90f120cf5226eae98e88ba283973881c79. I will apply this fix and then merge the PR.

If you are more attentive to these subtle problems, like dependencies and the NOTICE file, the PR review process could be a lot smoother.

Reply:
@aljoscha Thanks. Actually, I am really not familar with NOTICE file and things about license.

Reply:
Yes, it is easy to underestimate this. That is the problem, also for me. 😅